src/pybind/mgr/diskprediction/common/clusterdata.py

```diff
@@ -30,6 +30,8 @@ def differentiate(data1, data2):
     # >>> differentiate(*zip(times, values))
     0.5
     """
+    if float(data2[0] - data1[0]) == 0:
+        return 0
     return (data2[1] - data1[1]) / float(data2[0] - data1[0])

```

Comment:
Would be nice if you could merge both copies of `differentiate` into a new common file: `src/pybind/mgr/util.py`. Let's fight code duplication inside mgr modules.

Reply:
done.